notebook/tests/selenium/utils.py

```diff
@@ -33,6 +54,36 @@ def _wait_for(driver, locator_type, locator, timeout=10, visible=False, single=F
     return wait.until(conditional((locator_type, locator)))
 
 
+def _wait_for_multiple(driver, locator_type, locator, timeout, wait_for_n, visible=False):
+    """Waits until `wait_for_n` matching elements to be present (or visible).
+    Returns located elements when found.
+
+    Args:
+        driver: Selenium web driver instance
+        locator_type: type of locator (e.g. By.CSS_SELECTOR or By.TAG_NAME)
+        locator: name of tag, class, etc. to wait for
+        timeout: how long to wait for presence/visibility of element
+        wait_for_n: wait until this number of matching elements are present/visible
+        visible: if True, require that elements are not only present, but visible
+    """
+    wait = WebDriverWait(driver, timeout)
+
+    def multiple_found(driver):
+        try:
+            elements = driver.find_elements(locator_type, locator)
+        except WebDriverException as e:
+            raise e
+        if len(elements) < wait_for_n:
+            return False
+        if visible:
+            for element in elements:
+                if not element.is_displayed():
+                    return False
```

Comment:
I think this might need to be refined slightly. E.g. if I say `wait_for_tag('h2', wait_for_n=2, visible=True)` when there are 2 visible `h2` tags on the page and one invisible, it will never finish waiting.

Alternatively, we could leave out visibility from this code for now, and wait until there's a need for it.

Reply:
ah good point! I'll adjust to just look for the specified number visible.